### PR TITLE
Add CI check for non-ASCII characters

### PR DESCRIPTION
**Pull request by rswarbrick**
The first commit removes some non-ASCII characters from SystemVerilog code and our meson.build. I don't think this affects "editorial intent" at all: the changes are mostly just normalising punctuation and whitespace.

The second commit adds a CI check to make sure they don't creep back in.

### CODE REVIEW

File: ci/scripts/check-ascii.sh
Comment (mcy) [inline ci/scripts/check-ascii.sh:1]: Is it bad that I think this is a bit too complicated and maybe should just be a dumb python script

Comment (mcy) [inline ci/scripts/check-ascii.sh:1]: Is it bad that I think this is a bit too complicated and maybe should just be a dumb python script

File: ci/scripts/check-ascii.sh
Comment (asb) [reply ci/scripts/check-ascii.sh:26]: I'm a bit wary of having CI doing a repo-wide check for something that's not in the style guide. Though in this case it's probably not a big deal either way.

I think .py should be on the exclusion list as well (the style guide explicitly allows UTF-8).

File: hw/ip/kmac/rtl/keccak_2share.sv
Comment (sriyerg) [inline hw/ip/kmac/rtl/keccak_2share.sv:296]: Maybe I am missing something here, but what's non-ASCII about this change? Same below.

Comment (cdgori) [reply hw/ip/kmac/rtl/keccak_2share.sv:296]: I think the actual theta symbol is non-ASCII?

Comment (eunchan) [reply hw/ip/kmac/rtl/keccak_2share.sv:296]: Right, then consequence is whether it is OK to  put non-ASCII into verilog code or not. I intentionally added the symbol as the symbol is used in the SHA3 specification. So, this can be an exception?

Comment (sriyerg) [reply hw/ip/kmac/rtl/keccak_2share.sv:296]: Aah, nevermind.

File: ci/scripts/check-ascii.sh
Comment (rswarbrick) [reply ci/scripts/check-ascii.sh]: Good point about Python: I've just pushed a version that allows it. Honestly, I hadn't thought about this at all: it just so happens that our repo doesn't contain any non-ASCII Python at the moment!

@mcy: About C/C++: I don't really have an opinion about this, but let's discuss that as a separate PR. Maybe someone on the SW team could take the lead on that? Technically, it's pretty trivial (delete this block in the script and fix up the CI failures), but there's probably a proper discussion to be had.

File: ci/scripts/check-ascii.sh
Comment (rswarbrick) [reply ci/scripts/check-ascii.sh:1]: Well, this directory is full of simple bash scripts. If you want to rewrite them all in Python, I guess you can... :-)

File: hw/ip/kmac/rtl/keccak_2share.sv
Comment (rswarbrick) [reply hw/ip/kmac/rtl/keccak_2share.sv:296]: At the moment, the style guide doesn't contain any exceptions. I propose that we mandate conformance to our style guide. If we think there should be an exception (quite reasonable, in my opinion!), let's discuss it in the style guide first and then fix CI to match.

File: ci/scripts/check-ascii.sh
Comment (asb) [reply ci/scripts/check-ascii.sh]: I'd also be in favour of splitting out C/C++ style guide changes to another thread, so as not to block the .sv fixes.

**Review by asb — APPROVED** [review]
Thanks Rupert, this looks good to me but it would be good to get another LGTM before this lands. @mcy - are you happy to move discussions about porting CI scripts over to Python and changing the C/C++ style guide to another issue, and if so does this change look good to you too?

File: ci/scripts/check-ascii.sh
Comment (mcy) [reply ci/scripts/check-ascii.sh]: Sounds good. I'll file something once I can get to it.

File: ci/scripts/check-ascii.sh
Comment (cfrantz) [reply ci/scripts/check-ascii.sh:1]: Python is certainly nicer for the regex-preparing string manipulations in this script, but less so for the simplicity of the pipeline which performs the check (`git ls-files | grep filenames | xargs grep for non ascii`).

I'd suggest that the string-concatenation bit of this script should be put into a function, but even then, after all the bits are concatenated together, there is some final preparation of the regex which is slightly different between the two.

**Comment by rswarbrick** [comment]
I'm going to be on holiday for a week, so I'm not going to be following this. If we reach agreement, please feel free to amend and/or merge this without waiting for me.

File: hw/ip/kmac/rtl/keccak_2share.sv
Comment (imphil) [reply hw/ip/kmac/rtl/keccak_2share.sv:296]: The SV style guide says "Non-ASCII characters are forbidden.". That's not set in stone, but needs to be discussed in the style guide repo, not here.

**Review by imphil — APPROVED** [review]
Thanks Rupert! This PR enforces the agreed-on style for SV and the checking script matches the other checking scripts we are using today. Alternative approaches are possible, but beyond the scope of this PR. Style changes need to be discussed as part of a style guide change.

**Pull request merged**
